### PR TITLE
Add duckdb_param_logical_type

### PR DESCRIPTION
**Pull request by Giorgi**
Add a new function to the C API for getting prepared statement parameter logical type. Related to https://github.com/duckdb/duckdb/discussions/14476

### CODE REVIEW

**Comment by Giorgi** [comment]
The regression failure shouldn't be related to this PR

File: src/main/capi/prepared-c.cpp [hunk 1]
```diff
@@ -140,6 +140,25 @@ duckdb_type duckdb_param_type(duckdb_prepared_statement prepared_statement, idx_
 	return DUCKDB_TYPE_INVALID;
 }
 
+duckdb_logical_type duckdb_param_logical_type(duckdb_prepared_statement prepared_statement, idx_t param_idx) {
```
Comment (Tishj) [inline src/main/capi/prepared-c.cpp:143]: Thanks, can we replace the body of `duckdb_param_type` with this? Seeing as this is currently duplicating a lot of that logic

Inside `duckdb_param_type` we would only need to call `ConvertCPPTypeToC` on the result and then clean up the created `duckdb_logical_type`.

File: src/include/duckdb.h [hunk 2]
```diff
@@ -1471,6 +1471,17 @@ Returns `DUCKDB_TYPE_INVALID` if the parameter index is out of range or the stat
 */
 DUCKDB_API duckdb_type duckdb_param_type(duckdb_prepared_statement prepared_statement, idx_t param_idx);
 
+/*!
+Returns the parameter logical type for the parameter at the given index.
+
+Returns `nullptr` if the parameter index is out of range or the statement was not successfully prepared.
```
Comment (Tishj) [inline src/include/duckdb.h:1477]: Can we specify that the created `duckdb_logical_type` has to be cleaned up and with what function that has to happen?
Looking at `duckdb_column_logical_type`:
```
The return type of this call should be destroyed with `duckdb_destroy_logical_type`.
```

**Review by Tishj — APPROVED** [review]
Thanks, looks good to me now 👍

**Comment by Giorgi** [comment]
The failures don't seem to be related.

**Comment by Tishj** [comment]
Look like they're vcpkg related download failures

**Comment by Giorgi** [comment]
@Mytherin Can you rerun the failed jobs?

**Comment by Mytherin** [comment]
> @Mytherin Can you rerun the failed jobs?

I think this can be merged as-is. Thanks!

**Pull request merged**